manifests/check.pp

```diff
@@ -129,89 +129,39 @@
 ) {
 
-  validate_re($ensure, ['^present$', '^absent$'] )
-  if !is_bool($standalone) and  $standalone != 'absent' {
-    fail("sensu::check{${name}}: standalone must be a boolean or 'absent' (got: ${standalone})")
-  }
-  if $handle and !is_bool($handle) and  $handle != 'absent' {
-    fail("sensu::check{${name}}: handle must be a boolean or 'absent' (got: ${handle})")
-  }
-  if $publish and !is_bool($publish) and  $publish != 'absent' {
-    fail("sensu::check{${name}}: publish must be a boolean or 'absent' (got: ${publish})")
-  }
-  if $cron and !is_string($cron) and $cron != 'absent' {
-    fail("sensu::check{${name}}: cron must be a string or 'absent' (got: ${cron})")
-  }
-  if !is_integer($interval) and $interval != 'absent' {
-    fail("sensu::check{${name}}: interval must be an integer or 'absent' (got: ${interval})")
-  }
-  if $occurrences and !is_integer($occurrences) and $occurrences != 'absent' {
-    fail("sensu::check{${name}}: occurrences must be an integer or 'absent' (got: ${occurrences})")
-  }
-  if $refresh and !is_integer($refresh) and $refresh != 'absent' {
-    fail("sensu::check{${name}}: refresh must be an integer or 'absent' (got: ${refresh})")
-  }
-  if $low_flap_threshold and !is_integer($low_flap_threshold) and $low_flap_threshold != 'absent' {
-    fail("sensu::check{${name}}: low_flap_threshold must be an integer or 'absent' (got: ${low_flap_threshold})")
-  }
-  if $high_flap_threshold and !is_integer($high_flap_threshold) and $high_flap_threshold != 'absent' {
-    fail("sensu::check{${name}}: high_flap_threshold must be an integer or 'absent' (got: ${high_flap_threshold})")
-  }
-  if $timeout and !is_numeric($timeout) and $timeout != 'absent' {
-    fail("sensu::check{${name}}: timeout must be a numeric or 'absent' (got: ${timeout})")
-  }
-  if $ttl and !is_integer($ttl) and $ttl != 'absent' {
-    fail("sensu::check{${name}}: ttl must be an integer or 'absent' (got: ${ttl})")
-  }
-  if $dependencies and !is_array($dependencies) and !is_string($dependencies) {
-    fail("sensu::check{${name}}: dependencies must be an array or a string (got: ${dependencies})")
-  }
-  if $handlers and !is_array($handlers) and !is_string($handlers) {
-    fail("sensu::check{${name}}: handlers must be an array or a string (got: ${handlers})")
-  }
-  if $subscribers and !is_array($subscribers) and !is_string($subscribers) {
-    fail("sensu::check{${name}}: subscribers must be an array or a string (got: ${subscribers})")
-  }
-  if $subdue {
-    if is_hash($subdue) {
-      if !( has_key($subdue, 'days') and is_hash($subdue['days']) ){
-        fail("sensu::check{${name}}: subdue hash should have a proper format. (got: ${subdue}) See https://sensuapp.org/docs/latest/reference/checks.html#subdue-attributes")
-      }
-    } elsif !($subdue == 'absent') {
-      fail("sensu::check{${name}}: subdue must be a hash or 'absent' (got: ${subdue})")
+  if $subdue =~ Hash {
```

Comment:
The module has many type/providers though we are looking to minimize the number through using sensu_sorted_json().